x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -134,6 +169,53 @@ export function isModelPlotChartableForDetector(job: Job, detectorIndex: number)
   return isModelPlotChartable;
 }
 
+// Returns a reason to indicate why the job configuration is not supported
+// if the result is undefined, that means the single metric job should be viewable
+export function getSingleMetricViewerJobErrorMessage(job: CombinedJob): string | undefined {
+  let errorMessage;
+  if (!isTimeSeriesViewableJob(job)) {
+    errorMessage = i18n.translate('xpack.ml.timeSeriesJob.notViewableTimeSeriesJobMessage', {
+      defaultMessage: 'not a viewable time series job',
+    });
+  }
+  const hasDatafeed =
+    typeof job.datafeed_config === 'object' && Object.keys(job.datafeed_config).length > 0;
+  if (hasDatafeed) {
+    const aggs = getDatafeedAggregations(job.datafeed_config);
+    if (aggs !== undefined) {
+      const aggBucketsName = getAggregationBucketsName(aggs);
+      if (aggBucketsName !== undefined) {
+        // if datafeed has any nested terms aggregations at all
+        const aggregations = getAggregations<{ [key: string]: any }>(aggs[aggBucketsName]) ?? {};
+        const termsField = findAggField(aggregations, 'terms', true);
+        if (
+          termsField !== undefined &&
+          Object.keys(termsField).length > 1 &&
+          job.model_plot_config?.enabled !== true
+        ) {
+          errorMessage = i18n.translate('xpack.ml.timeSeriesJob.nestedTermsAggregationsMessage', {
+            defaultMessage:
+              'there are nested terms aggregations in the datafeed and model plot is disabled',
+          });
+        }
+
+        // if aggregation interval is different from bucket span
+        const datetimeBucket = aggs[aggBucketsName].date_histogram;
+        if (datetimeBucket?.fixed_interval !== job.analysis_config?.bucket_span) {
+          errorMessage = i18n.translate(
+            'xpack.ml.timeSeriesJob.varyingBucketSpanAggregationInterval',
+            {
+              defaultMessage:
+                'bucket span and aggregation interval is not the same for datafeed with aggregation fields',
```

Comment:
I think this could be clarified a bit. For example:
>                'the datafeed has aggregation fields and the aggregation interval is not the same as the bucket span',
or:
>                'bucket span and aggregation interval are not the same for a datafeed with aggregation fields', 
I prefer the first suggestion.

Reply:
My vote goes to the first suggestion from @lcawl

Reply:
Updated here https://github.com/elastic/kibana/pull/84594/commits/02a579be188ffa52c6836fb8a155cabcdb6b0598